torchmdnet/models/tensornet.py

```diff
@@ -236,16 +257,22 @@ def forward(
             edge_vec is not None
         ), "Distance module did not return directional information"
         # Distance module returns -1 for non-existing edges, to avoid having to resize the tensors when we want to ensure static shapes (for CUDA graphs) we make all non-existing edges pertain to a ghost atom
-        # Total charge q is a molecule-wise property. We transform it into an atom-wise property, with all atoms belonging to the same molecule being assigned the same charge q
-        if q is None:
-            q = torch.zeros_like(z, device=z.device, dtype=z.dtype)
-        else:
-            q = q[batch]
         zp = z
+        if extra_args is not None:
+            # we are assuming that extra args will be used, see model.py forward method how extra_args is passed
+            for label, t in extra_args.items():
+                # molecule wise --> atom wise 
+                if t.shape != z.shape:
+                    extra_args[label] = t[batch]
+            
```

Comment:
You should not assume everything will be used. You know the additional labels from the initializer.
This loop should be like:
```python
if self.additional_labels is not None:
     assert extra_args is not None
for label in self.additional_labels:
      assert label in extra_args, f"TensorNet expects {label} to be provided as part of extra_args"
      if extra_args[label].shape != z.shape:
         extra_args[label] = extra_args[label][batch]
```

Reply:
Yes, this was correctly implemented before when I was defining which extra_args should go to the forward pass in model.py. I will fix it, thanks!